src/Avalonia.Controls/TextBox.cs

```diff
@@ -1560,7 +1550,7 @@ protected override void OnPointerMoved(PointerEventArgs e)
             }
 
             // selection should not change during pointer move if the user right clicks
-            if (e.Pointer.Captured == _presenter && e.GetCurrentPoint(this).Properties.IsLeftButtonPressed)
+            if (e.Pointer.Captured == _presenter && e.Pointer.Type == PointerType.Mouse && e.GetCurrentPoint(this).Properties.IsLeftButtonPressed)
```

Comment:
Same as above, there's touch specific code inside this method that's can never be called anymore